packages/drawer/src/ux-modal-service.ts

```diff
@@ -0,0 +1,277 @@
+    }
+    if (options.moveToBodyTag === false) {
+      element.setAttribute('move-to-body-tag.bind', 'false');
+    }
+    if (options.overlayDismiss === false) {
+      element.setAttribute('overlay-dismiss', 'false');
+    }
+    if (options.keyboard !== undefined) {
+      bindingContext.keyboard = options.keyboard;
+      element.setAttribute('keyboard.bind', 'keyboard');
+    }
+    if (typeof options.restoreFocus === 'function') {
+      bindingContext.restoreFocus = options.restoreFocus;
+      element.setAttribute('restore-focus.bind', 'restoreFocus');
+    }
+    
+    element.setAttribute('host.bind', 'false');
+    
+    if (options.theme) {
+      bindingContext.theme = options.theme;
+      element.setAttribute('theme.bind', `theme`);
+    }
+    const compose = document.createElement('compose');
+    compose.setAttribute('view-model.ref', 'composeViewModel');
+    element.innerHTML = compose.outerHTML;
+    if (!options.host || options.host === 'body') {
+      options.host = document.body;
+    } else if (typeof options.host === 'string') {
+      options.host = document.querySelector(options.host) || document.body;
+    }
+    options.host.appendChild(element);
+    let childView = this.templatingEngine.enhance({ element: element, bindingContext: bindingContext });
+    
+    if (options.viewModel && bindingContext.composeViewModel) {
+      bindingContext.composeViewModel.viewModel = options.viewModel;
+    }
+    if (options.view && bindingContext.composeViewModel) {
+      bindingContext.composeViewModel.view = options.view;
+    }
+    if (options.model && bindingContext.composeViewModel) {
+      bindingContext.composeViewModel.model = options.model;
+    }
+
+    const controllers = (childView as any).controllers as Controller[];
+    
+    const drawer: UxDrawer =  controllers[0].viewModel as UxDrawer;
+    const drawerIndex = this.drawerIndex;
+    const whenClosed: Promise<ModalServiceResult> = new Promise((resolve) => {
+      this.eventAggregator.subscribeOnce(`drawer-${drawerIndex}-resolve`, (result: ModalServiceResult) => {
+        resolve(result);
+      });      
+    });
+
+    (drawer as UxDrawer & ModalServiceDrawer).whenClosed = () => {
+      return whenClosed;
+    }
+    bindingContext.dismiss = () => {
+      drawer.detached();
+      const parent = drawer.element.parentNode;
+      if (!parent) { return; }
+      parent.removeChild(drawer.element);
+      this.eventAggregator.publish(`drawer-${drawerIndex}-resolve`, {
+        wasCancelled: true,
+        output: null
+      });
+    }
+    bindingContext.ok = (event: CustomEvent) => {
+      drawer.detached();
+      const parent = drawer.element.parentNode;
+      if (!parent) { return; }
+      parent.removeChild(drawer.element);
+      this.eventAggregator.publish(`drawer-${drawerIndex}-resolve`, {
+        wasCancelled: false,
+        output: event.detail
+      });
+    }
+    return (drawer as UxDrawer & ModalServiceDrawer);
```

Comment:
nit: No need for parens here

Reply:
Can you explain what you mean here ?

The parent is use to `.removeChild()` - to remove the `ux-drawer` on both `dismiss()` and `ok()`

Did I miss something ?

Reply:
We can do:
```ts
    return drawer as UxDrawer & ModalServiceDrawer;
```